fix: linting tests on Form Editor

smbea opened the pull request:
#### Situation
* Tests for linting in the Form Editor check for any event being dispatched.

#### Problem
* If dispatches of other events that are not relevant to this test case are added, it will cause these tests to fail. 

#### Changes
* Tests assertions were modified to only consider events of type `lint-tab`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda/camunda-modeler?pullRequest=2417) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/camunda/camunda-modeler?pullRequest=2417) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda/camunda-modeler?pullRequest=2417) it.</sub>

barmac commented on client/src/app/tabs/form/__tests__/FormEditorSpec.js:
```diff
@@ -622,7 +622,14 @@ describe('<FormEditor>', function() {
+      });
+
+      // then
+      expect(calls.length).to.be.eql(1);
```
```suggestion
      expect(calls).to.have.lengthOf(1);
```

barmac replied:
So that for failure it says "expected calls to have length of 1" instead of "expected 2 to be equal 1".

smbea replied:
This is much cleaner. Thank you for the suggestion.

barmac commented on client/src/app/tabs/form/__tests__/FormEditorSpec.js:
```diff
@@ -622,7 +622,14 @@ describe('<FormEditor>', function() {
       const { form } = instance.getCached();
 
-      expect(onActionSpy).to.have.been.calledOnce;
+      var calls = onActionSpy.getCalls();
+
+      calls = calls.filter(function(call) {
+        return call.args[0] === 'lint-tab';
+      });
```
In Camunda Modeler, we use ES6 syntax:
```suggestion
      const calls = onActionSpy.getCalls()
        .filter(call => call.args[0] === 'lint-tab');
```

barmac commented on client/src/app/tabs/form/__tests__/FormEditorSpec.js:
```diff
@@ -709,8 +725,15 @@ describe('<FormEditor>', function() {
+
       // then
-      expect(onActionSpy).to.have.been.calledOnce;
+      expect(calls.length).to.be.eql(1);
+      expect(onActionSpy).to.have.been.calledWith('lint-tab');
```
The second expect statement can never fail on runtime. Either there is no call with `lint-tab`, which makes the test fail on the first statement, or it passes for both of them.

barmac commented:
Nice contribution which improves code quality.

barmac reviewed (approved):
Well done!

smbea commented:
* I committed the changes regarding the code consistency and the redundant assertion check.
* I rebased the branch and changed the commit message to be of type `test` instead of `fix`
